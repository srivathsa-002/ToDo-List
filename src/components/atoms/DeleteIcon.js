import React from 'react';
import DeleteOutlineOutlinedIcon from '@material-ui/icons/DeleteOutlineOutlined';
import { Tooltip, IconButton } from "@material-ui/core";

export default class DeleteIcon extends React.Component {
    render() {
        return (
            <Tooltip data-testid="delete" onClick={this.props.click} title="Delete Todo" placement={this.props.placement} arrow>
                <div>
                    <IconButton>
                        <DeleteOutlineOutlinedIcon />
                    </IconButton>
                </div>
            </Tooltip>
        );
    }
}