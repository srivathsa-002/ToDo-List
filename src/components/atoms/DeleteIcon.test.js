import React from "react";
import {render,fireEvent} from "@testing-library/react";
import DeleteIcon from "./DeleteIcon";
// import { faItalic } from "@fortawesome/free-solid-svg-icons";
import {create} from "react-test-renderer";

describe("DeleteIcon testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<DeleteIcon />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly", () => {
    const{getByTestId} = render(<DeleteIcon />)

    expect(getByTestId("delete")).toBeTruthy()
    // expect(queryByPlaceholderText("Todo...")).toBeTruthy()
})

// describe("Input value", () => {
//     it("updates on change", () => {
//         const {queryByPlaceholderText} = render(<InputBox />)
//         const inputBox = queryByPlaceholderText("Todo...")
//         fireEvent.change(inputBox, {target: {value: "test"}})
//         expect(inputBox.value).toBe("test")
//     })
// })
