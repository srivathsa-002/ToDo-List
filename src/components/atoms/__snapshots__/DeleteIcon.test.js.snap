// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteIcon testing Matches the snapshot 1`] = `
<div
  aria-describedby={null}
  className=""
  data-testid="delete"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Delete Todo"
>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4h-3.5z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
